src/main/java/org/spongepowered/common/mixin/core/world/MixinWorldServer.java

```diff
@@ -75,4 +84,37 @@ public void onPostInit(CallbackInfoReturnable<World> ci) {
         IMixinWorld world = (IMixinWorld) ci.getReturnValue();
         world.updateWorldGenerator();
     }
+
+    @Overwrite
+    public void updateAllPlayersSleepingFlag() {
+        this.allPlayersSleeping = false;
+        if (!this.playerEntities.isEmpty()) {
+            int ignoredPlayers = 0;
+            int sleepingPlayers = 0;
+            for (EntityPlayer entityPlayer : this.playerEntities) {
+                if (entityPlayer.isSpectator()
+                        || (entityPlayer instanceof Player && ((Player)entityPlayer).isSleepingIgnored())) {
```

Comment:
@SpongePowered/developers is this instanceof a valid check or the correct way to check this? Also I have a feeling that overriding the specific `isSpectator` call might be even better than an `@Inject` but it also seems a little bit like a hack.